docs/07.CODING-GUIDELINES.md

```diff
@@ -0,0 +1,610 @@
+# Coding style guidelines for the JerryScript project
+
+This text is a brief overview of JerryScript coding style.
+Each rule starts with a short description followed
+by several examples. We believe examples are better than
+long explanations. Please follow these guidelines when
+you submit a patch for review.
+
+## General rules
+
+* Indentation is two spaces.
+* Tab characters are not allowed.
+* Maximum line length is 120 characters long (excluding newline).
+* No trailing white spaces are allowed.
+* Run ```tools/run-tests.py --check-vera``` to check majority of the style rules.
+
+## Comments
+
+Only /* */ comments are allowed in JerryScript. Comments should
+be complete sentences, except for return value, field and
+argument descriptions (see these excpetions later). The text
+part of a comment should focus on explaing "why" rather than
+"what".
+
+```diff
++good
+```
+
+```c
+  /* A perfect comment. */
+
+  /* A perfect multiline
+   * comment. Each line should
+   * start with an asterisk. */
+```
+
+```diff
+-bad
+```
+
+```c
+  // Double slash comments are not allowed.
+
+  /* not a complete sentence */
+
+  /* A bad mulitiline
+     comment. */
+```
+
+All types, constants, and functions require a description in
+JerryScript. These comments should start with /**. The starting
+/** and ending */ must be in separate line.
+
+```diff
++good
+```
+
+```c
+/**
+ * A correct description.
+ */
+```
+
+```diff
+-bad
+```
+
+```c
+/** An incorrect description. */
+```
+
+## Preprocessor defines
+
+The name of a preprocessor macro must be an uppercase string
+and these macros must be preceeded by a description.
+Abbreviations are allowed but not preferred in new code.
+
+```diff
++good
+```
+
+```c
+/**
+ * Short description about the constant.
+ */
+#define JERRY_VALUE_SEVEN 7
+
+/**
+ * Short description about the macro function.
+ */
+#define JERRY_ADD_TWO_NUMBERS(arg1, arg2) \
+  ((arg1) + (arg2))
+
+/**
+ * Although this is correct, a reviewer might request
+ * to change NUMS to NUMBERS. Hence it is recommended
+ * to use NUMBERS in the first place.
+ */
+#define JERRY_ADD_TWO_NUMS(arg1, arg2) \
+  ((arg1) + (arg2))
+```
+
+```diff
+-bad
+```
+
+```c
+#define JERRY_CONSTANT_WITHOUT_DESCRIPTION 5
+
+#define JeRrY_mIxEd_CaSe_NaMe "str"
+```
+
+## Conditional preprocessor directives
+
+A comment is required after #else and #endif in JerryScript.
```

Comment:
IMHO: #else and #endif should be between backticks so they will appear as inline code. (Also applies to keywords and any other code related pars in the descriptions)